sysboundary/sysboundary.cpp

```diff
@@ -636,8 +636,11 @@ bool SysBoundary::classifyCells(dccrg::Dccrg<spatial_cell::SpatialCell, dccrg::C
 
                      technicalGrid.get(x, y, z)->sysBoundaryLayer = layer;
 
-                     if (layer > 2 && technicalGrid.get(x, y, z)->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) {
+                     if (layer > 2 && (technicalGrid.get(x,y,z)->sysBoundaryFlag == sysboundarytype::IONOSPHERE || 
+                                       technicalGrid.get(x,y,z)->sysBoundaryFlag == sysboundarytype::CONDUCTINGSPHERE)) {
                         technicalGrid.get(x, y, z)->sysBoundaryFlag = sysboundarytype::DO_NOT_COMPUTE;
+                     } else if (layer > 2 && technicalGrid.get(x, y, z)->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY) {
+                        technicalGrid.get(x, y, z)->sysBoundaryFlag = sysboundarytype::BOUNDARY_PADDING;
```

Comment:
I think I disagree with this logic.
At this stage (just before the loop) we have got the DCCRG types fed into `technicalGrid` at L514, so we can just use that and be done. No need for padding, upkeep the cell type and make sure the rest of the code is always checking layer as well as boundary type if we don't want to process cells that have boundary type && layer > 2.

Reply:
We just need to move L514 feeding to L590 after the mpiGrid update which includes the DO_NOT_COMPUTE assignments on the inside of the inner boundary.

Reply:
Or well indeed there was the original thing with the upstream wall – so move 514 to 590, and add a loop or modify the last loop, cells of type IONOSPHERE or CONDUCTINGSPHERE that have layer > 2 become PADDING and the other boundary types that are layer > 2 become DO_NOT_COMPUTE? I feel the confusion level rising...

Reply:
I think I'm just going to build a small testcase first, to see if this doesn't already do what we want now.

Reply:
I'll resolve this conversation, we ran several tests and my issues with the logic were that I had the opposite concept of which cells to flag differently. Works well like so too.